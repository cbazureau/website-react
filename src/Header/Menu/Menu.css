.Menu {
  margin: 0px;
  padding: 0px;
  display: flex;
  flex-direction: row;
  list-style: none;
  flex: 1;
  position: absolute;
  top: 0;
  width: 100%;
}

.Menu__link {
  padding: 4px 8px;
  margin: 10px 4px 6px 4px;
  background: none;
  display: block;
  border: none;
  text-decoration: none;
  font-size: 1.3rem;
  cursor: pointer;
  border-radius: 6px;
  color: #DDD;
  transition: background, color 0.5s;
}

.Menu__link:hover,
.Menu__link:focus {
  background: gray;
  color: white;
}



.Menu__link--cv {
  display: none;
  border: 1px solid gray;
  padding: 4px 12px;
  margin: 10px 4px 6px 4px;
}


@media only screen and (min-width: 650px) {
  .Menu__link--cv {
    display: block;
  }
}
