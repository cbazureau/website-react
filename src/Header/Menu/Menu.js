import React from 'react';
import './Menu.css';

const Menu = ({ onSectionChange }) => (
  <ul className="Menu">
    <li key="link-home"><button className="Menu__link" onClick={onSectionChange("current")}>Poste Actuel</button></li>
    <li key="link-xp"><button className="Menu__link" onClick={onSectionChange("xp")}>Expertise</button></li>
    <li key="link-jobs"><button className="Menu__link" onClick={onSectionChange("jobs")}>Carri&egrave;re</button></li>
  </ul>
);

export default Menu;
