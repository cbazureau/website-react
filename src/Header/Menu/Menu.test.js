import React from 'react';
import renderer from 'react-test-renderer';
import Menu from './Menu';

const onSectionChange = () => () => null;

describe('Menu component', () => {
  it('renders with props', () => {
    const component = renderer.create(<Menu onSectionChange={onSectionChange} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    const component = renderer.create(<Menu />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
