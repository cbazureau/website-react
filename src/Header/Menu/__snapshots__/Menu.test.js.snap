// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders with props 1`] = `
<ul
  className="Menu"
>
  <li>
    <button
      className="Menu__link"
      onClick={[Function]}
    >
      Current Job
    </button>
  </li>
  <li>
    <button
      className="Menu__link"
      onClick={[Function]}
    >
      Expertise
    </button>
  </li>
  <li>
    <button
      className="Menu__link"
      onClick={[Function]}
    >
      Career
    </button>
  </li>
</ul>
`;

exports[`Menu component renders without crashing 1`] = `
<ul
  className="Menu"
>
  <li>
    <button
      className="Menu__link"
    >
      Current Job
    </button>
  </li>
  <li>
    <button
      className="Menu__link"
    >
      Expertise
    </button>
  </li>
  <li>
    <button
      className="Menu__link"
    >
      Career
    </button>
  </li>
</ul>
`;
